### ISSUE TITLE
[MCOMPILER-434] Annotation processor exceptions in forked Javac are ignored

### ISSUE THREAD

**Issue by olamy** [olamy/maven-compiler-plugin#374]
**[Dan Berindei](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=dan.berindei)** opened **[MCOMPILER-434](https://issues.apache.org/jira/browse/MCOMPILER-434?redirect=false)** and commented

A POM with an annotation processor missing a dependency will fail to compile, but maven-compiler-plugin will not print the exception.

See https://github.com/codehaus-plexus/plexus-compiler/issues/106

Reproducer here: https://github.com/danberindei/maven-compiler-test/tree/annotation_processor_exception

---

**Affects:** 3.8.1

2 votes, 4 watchers

**Comment by olamy** [comment]
**[Delany](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=delany)** commented

This is resolved with maven-compiler-plugin:3.10.1

```
[ERROR] java.lang.NoClassDefFoundError: javax/annotation/Generated
[ERROR]         at org.infinispan.protostream.annotations.impl.processor.AutoProtoSchemaBuilderAnnotationProcessor.addGeneratedClassHeader(AutoProtoSchemaBuilderAnnotationProcessor.java:518)
[ERROR]         at org.infinispan.protostream.annotations.impl.processor.MarshallerSourceCodeGenerator.generateMessageMarshaller(MarshallerSourceCodeGenerator.java:156)
[ERROR]         at org.infinispan.protostream.annotations.impl.processor.MarshallerSourceCodeGenerator.generateMarshaller(MarshallerSourceCodeGenerator.java:56)
[ERROR]         at org.infinispan.protostream.annotations.impl.BaseProtoSchemaGenerator.generateMarshallers(BaseProtoSchemaGenerator.java:204)
[ERROR]         at org.infinispan.protostream.annotations.impl.BaseProtoSchemaGenerator.generateAndRegister(BaseProtoSchemaGenerator.java:175)
[ERROR]         at org.infinispan.protostream.annotations.impl.processor.AutoProtoSchemaBuilderAnnotationProcessor.processClass(AutoProtoSchemaBuilderAnnotationProcessor.java:320)
[ERROR]         at org.infinispan.protostream.annotations.impl.processor.AutoProtoSchemaBuilderAnnotationProcessor.processElement(AutoProtoSchemaBuilderAnnotationProcessor.java:251)
[ERROR]         at org.infinispan.protostream.annotations.impl.processor.AutoProtoSchemaBuilderAnnotationProcessor.process(AutoProtoSchemaBuilderAnnotationProcessor.java:178)
[ERROR]         at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment.callProcessor(JavacProcessingEnvironment.java:985)
[ERROR]         at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment.discoverAndRunProcs(JavacProcessingEnvironment.java:901)
[ERROR]         at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment$Round.run(JavacProcessingEnvironment.java:1227)
[ERROR]         at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment.doProcessing(JavacProcessingEnvironment.java:1340)
[ERROR]         at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.processAnnotations(JavaCompiler.java:1258)
[ERROR]         at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:936)
[ERROR]         at jdk.compiler/com.sun.tools.javac.main.Main.compile(Main.java:311)
[ERROR]         at jdk.compiler/com.sun.tools.javac.main.Main.compile(Main.java:170)
[ERROR]         at jdk.compiler/com.sun.tools.javac.Main.compile(Main.java:57)
[ERROR]         at jdk.compiler/com.sun.tools.javac.Main.main(Main.java:43)
[ERROR] Caused by: java.lang.ClassNotFoundException: javax.annotation.Generated
[ERROR]         at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:476)
[ERROR]         at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:589)
[ERROR]         at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
[ERROR]         ... 18 more
```

**Comment by olamy** [comment]
**[Olivier Lamy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=olamy)** commented

fixed by plexus-compiler
